### ISSUE TITLE
Bug: [no-unnecessary-qualifier] False positive with enum/namespace merge

### ISSUE THREAD

**Issue by rchl** [typescript-eslint/typescript-eslint#5664]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.8.3&sourceType=module&code=KYDwDg9gTgLgBMAdgVwLZwCpQIYGNhwDeAUHHAPIBmlANKXALLADOz2A5i3WQGrBQAjCM2B0AvsWKhIsOImyoWYPASwqi9adHiVkiXDACWERHEpQIqAMowohxOwAUAN2wAbZMABccZrfvsAJQ+avgaZGSuHgQAvHBRngB0MBAAMhAA7vwAwtgijoEA3PRkzBmGMLgAFnAu7p6B4RFkuHkEAOQQ1O1eJc1wUMAwyFCmVJTF-XCtInDtiqwcLD19zYPDo5g4+IlMi5zMk-0zHc78QiIrU2TrI6ahwIl8gsLAR80AJsCU2MhuML1rgMhnctipEuN3nAJGQJGIgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0AjrCGMAZpUTR0UBdA6LwYAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
export enum Trace {
  Off,
  Messages,
  Verbose,
}

export namespace Trace {
  export function fromString(value: string): Trace {
    value = value.toLowerCase();
    switch (value) {
      case 'off':
        return Off;
      case 'messages':
        return Trace.Messages;
      case 'verbose':
        return Trace.Verbose;
      default:
        return Trace.Off;
    }
  }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unnecessary-qualifier": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Expected that the rule doesn't complain about the qualifier `Trace` being used as without it the Typescript itself produces an error `Cannot find name ...`.

### Actual Result

Errors raised when `Trace` qualifier is used.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.37.0` |
| `@typescript-eslint/parser`            | `5.37.0` |
| `TypeScript`                           | `4.8.3` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `16` |

**Comment by bradzacher** [comment]
It's rare that we see people declaration merging with an enum!
Definitely a bug.

**Comment by RealShadowNova** [comment]
I'm currently facing this bug, any ETA on it's fix?
